modules/xmpp/JingleSessionPC.js

```diff
@@ -1877,21 +1868,15 @@ export default class JingleSessionPC extends JingleSession {
                 = isAdd
                     ? this._processRemoteAddSource(addOrRemoveSsrcInfo)
                     : this._processRemoteRemoveSource(addOrRemoveSsrcInfo);
-
-            // Add a workaround for a bug in Chrome (unified plan) for p2p connection. When the media direction on
-            // the transceiver goes from "inactive" (both users join muted) to "recvonly" (peer unmutes), the browser
-            // doesn't seem to create a decoder if the signaling state changes from "have-local-offer" to "stable".
-            // Therefore, initiate a responder renegotiate even if the endpoint is the offerer to workaround this issue.
-            // TODO - open a chrome bug and update the comments.
             const remoteDescription = new RTCSessionDescription({
                 type: 'offer',
                 sdp: newRemoteSdp.raw
             });
-            const promise = isAdd && this.usesUnifiedPlan && this.isP2P && browser.isChromiumBased()
-                ? this._responderRenegotiate(remoteDescription)
-                : this._renegotiate(newRemoteSdp.raw);
 
-            promise.then(() => {
+            // Always initiate a sRD->cA->sLD cycle when a remote source is added or removed irrespective of whether
```

Comment:
Is this safe to add right now? Note that mobile 2 mobile will still use p2p in plan B mode.

Reply:
Yes it, it should have always been implemented this way. Without this change, on p2p offerer, we call createOffer, setLocalDescription and then send out the offer to the peer. When the response is received, we again run the same cycle of cO->sLD->sRD. So we are calling createOffer (on receiving answer) when the peerconnection state is "have-local-offer" which is not correct. Plan-b is more forgiving and we don't see the rendering issues there but unified-plan is more strict. I don't see issues doing it this way for plan-b p2p.